jetty-ee10/jetty-ee10-proxy/src/main/java/org/eclipse/jetty/ee10/proxy/BalancerServlet.java

```diff
@@ -217,7 +217,8 @@ protected String filterServerResponseHeader(HttpServletRequest request, Response
             URI locationURI = URI.create(headerValue).normalize();
             if (locationURI.isAbsolute() && isBackendLocation(locationURI))
             {
-                StringBuilder newURI = URIUtil.newURIBuilder(request.getScheme(), request.getServerName(), request.getServerPort());
+                StringBuilder newURI = new StringBuilder();
+                HttpScheme.appendNormalizedUri(newURI, request.getScheme(), request.getServerName(), request.getServerPort());
```

Comment:
The code below this line adds path, query and fragment, so you can just use the new method with more parameters.